### PR TITLE
Ensure that Kotlin subclass of QuarkusApplication works properly

### LINKED ISSUE #30054: 2.15.1.Final breaks command mode with main class extends from `QuarkusApplication` in kotlin
### Describe the bug

Pretty sure it's from this pull https://github.com/quarkusio/quarkus/pull/29755.
`sanitizeMainClassName` method overlooks that main class extends from `QuarkusApplication` still compiled as original name (no postfix Kt)
https://github.com/quarkusio/quarkus/blob/6776f0deeb8537c5207ddb5ad08e7b3c5db83722/core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java#L416-L427

---

**With**
```kt
@QuarkusMain(name = "my-main")
class MyMainClass : QuarkusApplication {
  override fun run(vararg args: String?): Int {
    // omitted
  }
}
```
**Running**
```
./gradlew quarkusDev -Dquarkus.package.main-class=my-main
```
**would produces following error**
```
Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.MainClassBuildStep#mainClassBuildStep threw an exception: java.lang.IllegalArgumentException: The supplied 'main-class' value of '...MyMainClassKt' does not correspond to either a fully qualified class name or a matching 'name' field of one of the '@QuarkusMain' annotations
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #30054

### CODE REVIEW

File: integration-tests/gradle/src/main/resources/basic-kotlin-application-project/src/main/kotlin/org/acme/MyMainClass.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package `basic-kotlin-application-project`.src.main.kotlin.org.acme
+
+import io.quarkus.runtime.Quarkus
+import io.quarkus.runtime.QuarkusApplication
+import io.quarkus.runtime.annotations.QuarkusMain
+
+@QuarkusMain(name = "my-main")
+class MyMainClass : QuarkusApplication {
```
Comment (evanchooly) [inline integration-tests/gradle/src/main/resources/basic-kotlin-application-project/src/main/kotlin/org/acme/MyMainClass.kt:8]: This all looks fine to me, in general, but I'm curious why this wouldn't have the super class in the check above.  or is that check intended for top level functions and this test just doesn't exercise that code path?

**Review by evanchooly — APPROVED** [review]
Looks like maybe another test might be needed if i'm reading this code correctly but otherwise seems fine.  This test is in the gradle IT tree but it doesn't look like there's anything specifically tied to gradle here so there's nothing to be done for maven, yes?

File: integration-tests/gradle/src/main/resources/basic-kotlin-application-project/src/main/kotlin/org/acme/MyMainClass.kt (hunk 1 above)
Comment (geoand) [reply integration-tests/gradle/src/main/resources/basic-kotlin-application-project/src/main/kotlin/org/acme/MyMainClass.kt:8]: Right. I could add another level of superclass in order to exercise the recursive lookup.

**Comment by geoand** [comment]
> This test is in the gradle IT tree but it doesn't look like there's anything specifically tied to gradle here so there's nothing to be done for maven, yes?

Exactly. The reason I added it to the gradle tests is because we didn't have any more Kotlin modules in the integration-tests

**Pull request merged**